hal/src/samd11/timer.rs

```diff
@@ -174,6 +174,20 @@ impl TimerParams {
     {
         let timeout = timeout.into();
         let ticks: u32 = src_freq / timeout.0.max(1);
+        Self::new_from_ticks(ticks)
+    }
+
+    pub fn new_us<T>(timeout: T, src_freq: Hertz) -> Self
+    where
+        T: Into<Microseconds>,
+    {
+        let timeout = timeout.into();
+        let ticks: u32 = (timeout.0 as u64 * src_freq.0 as u64 / 1_000_000_u64) as u32;
```

Comment:
I went back and forth on this one. I initially had:
```rust
let src_frq_mhz: MegaHertz = src_freq.into();
let ticks: u32 = timeout.0 * src_freq_mhz.0;
```
which avoids `u64` shenanigans (to avoid overflow), but for any source frequencies less than 1 MHz, or even non-multiples of 1 MHz, this introduces rounding errors. Personally, doing a little bit of `u64` math isn't too bad, but curious what people think. Currently I don't think it's even possible to have a non-48 MHz clock :p